gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -1499,6 +1499,28 @@ gst_tensors_config_from_structure (GstTensorsConfig * config,
   return TRUE;
 }
 
+/**
+ * @brief Parse caps and set tensors config (for other/tensors)
+ * @param config tensors config structure to be filled
+ * @param caps incoming capability
+ * @return TRUE/FALSE (if successfully configured, return TRUE)
+ */
+gboolean
+gst_tensors_config_from_cap (GstTensorsConfig * config, GstCaps * caps)
+{
+  GstStructure *structure;
+
+  if (!gst_caps_is_fixed (caps)) {
+    nns_logw ("GstCaps is not fixed");
+    return FALSE;
+  }
+
+  structure = gst_caps_get_structure (caps, 0);
+  gst_tensors_config_from_structure (config, structure);
```

Comment:
You may update other elements and utils also :)
e.g., gst_tensor_aggregator_parse_caps() gst_tensordec_configure() gst_tensor_sparse_dec_sink_event() gst_tensor_trainer_sink_event() gst_tensor_transform_read_caps() gst_tensor_transform_transform_caps() gst_tensor_time_sync_buffer_from_collectpad() gst_tensor_sink_grpc_setcaps() gst_tensor_src_grpc_set_caps()